### PR TITLE
Fix/seeding

### LINKED ISSUE #1656: Faker returns one value when using multiple locales
* Faker version: 13.11
* OS: Mac OS Monterey
Computer: MacBook Pro (14-inch, 2021)

Brief summary of the issue goes here.
If using multiple locales, faker only makes one choice. This only happens when you select multiple locales

### Steps to reproduce

```
from faker import Faker

fake = Faker(["en_GB", "fr_FR", "en_IN"])
Faker.seed(0)
for _ in range(5):
    print(fake.date_time_this_year())

```

### Expected behavior
returns
```
2022-03-16 19:22:23
2022-03-23 16:00:20
2022-01-08 20:40:15
2022-02-20 06:38:22
2022-04-10 06:42:46
```

### Actual behavior
returns
```
2022-03-23 16:00:20
2022-03-23 16:00:20
2022-03-23 16:00:20
2022-03-23 16:00:20
2022-03-23 16:00:20

```

### PR DESCRIPTION
**Pull request by johntmyers**
fixes #1656 

### What does this change

This removes re-seeding the `Random()` instance on every call to `_select_factory()` which gets called every time a faker method is used. This code was only reached when both conditions are true:

- A seed is set via `Faker.seed()`
- More than one locale is specified

### What was wrong

Basically same as above, it is unnecessary to set the `Random()` instance's seed value every time a faker method is used. The `Random()` instance is seeded when making the call to `Faker.seed()` which is sufficient for making sure that the order of the locales selected is the same between subsequent instances of `Faker`

### How this fixes it

Remove the seeding that isn't needed and modified a test that asserts:

- Generating values when a seed is used and multiple locales are used does not generate the same value over and over
- Calling `Faker.seed()` and using a `Faker` instance in an identical way will produce the same fake output each time

Fixes #...

### CODE REVIEW

File: tests/test_proxy.py [hunk 1]
```diff
@@ -130,11 +130,17 @@ def test_seed_classmethod(self):
 
     def test_seed_class_locales(self):
         Faker.seed(2043)
+        count = 5
         fake = Faker(["en_GB", "fr_FR", "en_IN"])
-        name = fake.name()
+        first_list = [fake.name() for _ in range(count)]
+        assert len(set(first_list)) == count
```
Comment (fcurella) [inline tests/test_proxy.py:136]: what's the reason for casting to a `set`?

Comment (fcurella) [reply tests/test_proxy.py:136]: got it 👍🏼  Would you add a comment above the assertion?

**Comment by fcurella** [comment]
@johntmyers You can run `make lint` to format the code with our settings.

File: tests/test_proxy.py (hunk 1 above)
Comment (johntmyers) [reply tests/test_proxy.py:138]: done

**Comment by johntmyers** [comment]
@fcurella everything is passing now

**Review by fcurella — APPROVED** [review]
Thank you!

**Pull request merged**